[Bug]: Image length has a limit to respect

### Is there an existing issue already for this bug?

- [x] I have searched for an existing issue, and could not find anything. I believe this is a new bug.

### I have read the troubleshooting guide

- [x] I have read the troubleshooting guide and I think this is a new bug.

### I am running a supported version of CloudNativePG

- [x] I have read the troubleshooting guide and I think this is a new bug.

### Contact Details

_No response_

### Version

1.28 (latest patch)

### What version of Kubernetes are you using?

1.35

### What is your Kubernetes environment?

Self-managed: kind (evaluation)

### How did you install the operator?

YAML manifest

### What happened?

The workflow that generates the bundle uses the branch name to create the tag, the branch names must of the time may be really long, like in this case https://github.com/cloudnative-pg/cloudnative-pg/pull/10154 and this generates a really long tag that breaks the spec (https://github.com/opencontainers/distribution-spec/blob/main/spec.md#pulling-manifests) 

We should have a limit of 50 characters or similar for the tag to avoid having these issue, at least for the OLM tag that has this problem:
```
2026/03/06 08:25:16 bundle/manifests/cloudnative-pg.clusterserviceversion.yaml - Replaced pullspec for relatedImage cnpg: {ghcr.io cloudnative-pg cloudnative-pg-testing renovate-main-https-github.com-redhat-openshift-ecosystem-community-operators-pipeline.git-digest-ubi9} -> {ghcr.io cloudnative-pg cloudnative-pg-testing sha256:2e657bca4f4277182b2e8827867584fe7fa80b89390bf0281e11d13e6b494d2d}
2026/03/06 08:25:16 bundle/manifests/cloudnative-pg.clusterserviceversion.yaml - Replaced pullspec for container manager: {ghcr.io cloudnative-pg cloudnative-pg-testing renovate-main-https-github.com-redhat-openshift-ecosystem-community-operators-pipeline.git-digest-ubi9} -> {ghcr.io cloudnative-pg cloudnative-pg-testing sha256:2e657bca4f4277182b2e8827867584fe7fa80b89390bf0281e11d13e6b494d2d}
2026/03/06 08:25:16 bundle/manifests/cloudnative-pg.clusterserviceversion.yaml - Replaced pullspec for cnpg var: {ghcr.io cloudnative-pg cloudnative-pg-testing renovate-main-https-github.com-redhat-openshift-ecosystem-community-operators-pipeline.git-digest-ubi9} -> {ghcr.io cloudnative-pg cloudnative-pg-testing sha256:2e657bca4f4277182b2e8827867584fe7fa80b89390bf0281e11d13e6b494d2d}
2026/03/06 08:25:16 bundle/manifests/cloudnative-pg.clusterserviceversion.yaml - Replaced pullspec for annotation cloudnative-pg-testing-renovate-main-https-github.com-redhat-openshift-ecosystem-community-operators-pipeline.git-digest-ubi9-annotation: {ghcr.io cloudnative-pg cloudnative-pg-testing renovate-main-https-github.com-redhat-openshift-ecosystem-community-operators-pipeline.git-digest-ubi9} -> {ghcr.io cloudnative-pg cloudnative-pg-testing sha256:2e657bca4f4277182b2e8827867584fe7fa80b89390bf0281e11d13e6b494d2d}
panic: runtime error: slice bounds out of range [:148] with length 117

goroutine 1 [running]:
github.com/operator-framework/operator-manifest-tools/pkg/pullspec.(*Annotation).Image(0xc000ed3620?)
	/home/runner/go/pkg/mod/github.com/operator-framework/operator-manifest-tools@v0.10.0/pkg/pullspec/pullspec.go:211 +0xb1
github.com/operator-framework/operator-manifest-tools/pkg/pullspec.(*OperatorCSV).ReplacePullSpecs(0xc00010e640, 0xc000ed34a0)
	/home/runner/go/pkg/mod/github.com/operator-framework/operator-manifest-tools@v0.10.0/pkg/pullspec/pullspec.go:463 +0x8d
github.com/operator-framework/operator-manifest-tools/pkg/pullspec.(*OperatorCSV).ReplacePullSpecsEverywhere(0xc00010e640, 0xc000ed34a0)
	/home/runner/go/pkg/mod/github.com/operator-framework/operator-manifest-tools@v0.10.0/pkg/pullspec/pullspec.go:477 +0x32
github.com/operator-framework/operator-manifest-tools/pkg/image.Replace({0xc00091d7b8, 0x1, 0xc000c75f40?}, 0xc000ed34a0)
	/home/runner/go/pkg/mod/github.com/operator-framework/operator-manifest-tools@v0.10.0/pkg/image/replace.go:35 +0x65
github.com/operator-framework/operator-manifest-tools/pkg/image.Pin({0x357f140, 0xc0014a0000}, {0xc00091d7b8, 0x1, 0x1})
	/home/runner/go/pkg/mod/github.com/operator-framework/operator-manifest-tools@v0.10.0/pkg/image/pin.go:19 +0x7e
github.com/operator-framework/operator-sdk/internal/cmd/operator-sdk/generate/bundle.bundleCmd.pinImages({0x1, 0x1, {0x7ffd5772562f, 0xd}, {0x0, 0x0}, {0x3023f73, 0x6}, {0xc0006a7d20, 0x10}, ...}, ...)
	internal/cmd/operator-sdk/generate/bundle/bundle.go:320 +0x11e
github.com/operator-framework/operator-sdk/internal/cmd/operator-sdk/generate/bundle.bundleCmd.runManifests({0x1, 0x1, {0x7ffd5772562f, 0xd}, {0x0, 0x0}, {0x3023f73, 0x6}, {0xc0006a7d20, 0x10}, ...})
	internal/cmd/operator-sdk/generate/bundle/bundle.go:233 +0xb7b
github.com/operator-framework/operator-sdk/internal/cmd/operator-sdk/generate/bundle.NewCmd.func1(0xc00045b508?, {0xc0008e4f70?, 0x0?, 0xd?})
	internal/cmd/operator-sdk/generate/bundle/cmd.go:96 +0x3c5
github.com/spf13/cobra.(*Command).execute(0xc00045b508, {0xc0008e4ea0, 0xd, 0xd})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1015 +0xaaa
github.com/spf13/cobra.(*Command).ExecuteC(0xc000871b08)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1148 +0x46f
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1071
sigs.k8s.io/kubebuilder/v4/pkg/cli.CLI.Run({{0x303acaa, 0xc}, {0xc0003fe0b0, 0xa9}, {0x0, 0x0}, {0x3101a1e, 0x37}, 0xc000851ef0, 0xc000851f20, ...})
	/home/runner/go/pkg/mod/sigs.k8s.io/kubebuilder/v4@v4.6.0/pkg/cli/cli.go:473 +0x1b
github.com/operator-framework/operator-sdk/internal/cmd/operator-sdk/cli.Run()
	internal/cmd/operator-sdk/cli/cli.go:66 +0x65
main.main()
	cmd/operator-sdk/main.go:28 +0x13
make: *** [Makefile:177: olm-bundle] Error 2
```

### Cluster resource

```shell

```

### Relevant log output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/.github/workflows/continuous-integration.yml
+++ b/.github/workflows/continuous-integration.yml
@@ -464,7 +464,7 @@ jobs:
           fi
 
           # extract tag from branch name
-          tag_name=$(echo "$branch_name" | tr / -)
+          tag_name=$(echo "$branch_name" | tr / - | head -c 50)
 
           echo "DATE=${commit_date}" >> $GITHUB_ENV
           echo "VERSION=${commit_version}" >> $GITHUB_ENV

